### PR TITLE
chore: update client to pass jwt token if present

### PR DESCRIPTION
**Pull request by davidstanley01**
WF-10

This change allows for an optional token to be set and used for the duration the client is instantiated.

Commit [c5f6412](https://github.com/Watchfulio/watchful-py/commit/c5f641278f1924175f7670d806bcc4f1c13730f1) contains fixes due to failing black check.

This PR is needed before https://github.com/Watchfulio/watchful/pull/2964 can be merged

### CODE REVIEW

File: src/watchful/__about__.py [hunk 1]
```diff
@@ -1,3 +1,3 @@
 """Package meta information for watchful."""
 
-__version__ = "3.7.0"
+__version__ = "3.8.0"
```
Comment (rockstar) [inline src/watchful/__about__.py:3]: Let's not bump versions or release anything here. We can pin the git rev id in our tests to use this updated version.

File: src/watchful/__init__.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 """Watchful python library"""
+
```
Comment (rockstar) [inline src/watchful/__init__.py:2]: Did formatting change this?

File: src/watchful/attributes.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 """
 This script provides the functions required for data enrichment.
 """
+
```
Comment (rockstar) [inline src/watchful/attributes.py:4]: Yeah, I'd be curious what formatting changed this, but I'd also like to see these reverted. Formatting-only changes to files should not live in patches that have actual functionality changes.

File: src/watchful/client.py [hunk 4]
```diff
@@ -37,6 +37,7 @@
 HOST: str = "localhost"
 PORT: Optional[str] = "9002"
 API_TIMEOUT_SEC: int = 600
+TOKEN: Optional[str] = None
```
Comment (rockstar) [inline src/watchful/client.py:40]: I'm not really a fan of making this optional, but I get that this api leaves a lot to be desired.

File: src/watchful/client.py [hunk 5]
```diff
@@ -362,17 +366,41 @@ def external(
...
+    TOKEN = token
+
+
+def clear_token() -> None:
+    """
+    This function clears the global ``TOKEN`` value.
+    """
+
+    global TOKEN
+    TOKEN = None
```
Comment (rockstar) [inline src/watchful/client.py:403]: Can you explain a situation where we would need these two functions, and couldn't just use `external` from above?

File: src/watchful/enrich.py [hunk 6]
```diff
@@ -2,6 +2,7 @@
 This script is run on the Python command line to execute data enrichment. If a
 custom enricher is used, then the :func:`enrich_dataset` function is used.
 """
+
```
Comment (rockstar) [inline src/watchful/enrich.py:5]: Can you please revert this?

File: src/watchful/enricher.py [hunk 7]
```diff
@@ -5,6 +5,7 @@
 https://github.com/Watchfulio/watchful-py/blob/main/examples/enrichment_intro.ipynb
 for a tutorial on how to implement your custom enricher class.
 """
+
```
Comment (rockstar) [inline src/watchful/enricher.py:8]: This one too.

**Review by rockstar — COMMENTED** [review]
Thanks for this. Have a couple of questions, and a few requests.

File: src/watchful/__about__.py (hunk 1 above)
Comment (davidstanley01) [reply src/watchful/__about__.py:3]: that's a good idea. Next commit will revert this.

File: src/watchful/__init__.py (hunk 2 above)
Comment (davidstanley01) [reply src/watchful/__init__.py:2]: Yeah. `hatch run check` was failing in the black step saying it'd reformat all of this.

File: src/watchful/attributes.py (hunk 3 above)
Comment (davidstanley01) [reply src/watchful/attributes.py:4]: see above

File: src/watchful/client.py (hunk 4 above)
Comment (davidstanley01) [reply src/watchful/client.py:40]: I don't like anything about this api, either

File: src/watchful/client.py (hunk 5 above)
Comment (davidstanley01) [reply src/watchful/client.py:403]: we could, sure - `client.external(token = None)` but I thought the `set_token` and `clear_token` functions, though simple, were much clearer to the reader. Purely for convenience.

File: src/watchful/attributes.py (hunk 3 above)
Comment (davidstanley01) [reply src/watchful/attributes.py]: done

File: src/watchful/__init__.py (hunk 2 above)
Comment (davidstanley01) [reply src/watchful/__init__.py]: done

File: src/watchful/enrich.py (hunk 6 above)
Comment (davidstanley01) [reply src/watchful/enrich.py]: done

File: src/watchful/enricher.py (hunk 7 above)
Comment (davidstanley01) [reply src/watchful/enricher.py]: done

File: src/watchful/client.py (hunk 5 above)
Comment (rockstar) [reply src/watchful/client.py:403]: As this module is almost entirely for running tests, I'm not too worried about convenience. The reason I bring it up is that I can't see a case where tests would use these, and so I think we can drop.

Comment (davidstanley01) [reply src/watchful/client.py]: removed

**Review by rockstar — APPROVED** [review]

**Pull request merged**